Community/Ephemeris Time of Day/automation/jsr223/python/community/ephem_tod/ephem_tod.py

```diff
@@ -0,0 +1,436 @@
+                  special days, or just define special days you want to treat
+                  differently. Defined with the following metadata:
+                      - ToD="day"[type="custom", file="/openhab/conf/services/custom1.xml"]
+
+    For example, to define times of day that covers the weekend, weekdays, and
+    a custom set of bank holidays you might have the following Groups.
+
+        Group Ephem_TimeOfDay_StartTimes
+        Group Ephem_Weekday (Ephem_TimeOfDay_StartTimes) { ToD="day"[type="weekday"] }
+        Group Ephem_Weekend (Ephem_TimeOfDay_StartTimes) { ToD="day"[type="weekend"] }
+        Group Ephem_Custom1 (Ephem_TimeOfDay_StartTimes) { ToD="day"[type="custom", file="/openhab/conf/services/custom1.xml"] }
+
+    You must be sure to define a Group for every type of day about. It can be
+    handy to define the default Group to ensure that happens. The day types are
+    checked in the order listed above. So, for example, if you have all of the
+    above types defined, on a day that is a work day (custom dataset) that is a
+    custom holiday, the custom holiday Group will be chosen.
+
+    You can define multiple Groups for dayset and custom. If you define more
+    than one Group for any of the rest, only the first one found will be used.
+
+    Now that you have the days defined, you must define a set of DateTime Items
+    which hold the start times for each time of day state used on that day and
+    add those Item to the given day's Group. For example, the start times for
+    the weekend would be added to the Ephem_Weekend Group above.
+
+    The values stored in the Items can come from anywhere but the two most
+    common will be from Astro or statically defined. For Astro, just link the
+    Item to the appropriate Astro Channel. For statically defined the time is
+    defined and the Item initialized through metadata.
+
+    The expected metadata is : Tod="init"[start_time="HH:MM:SS:MS", tod_state="EXAMPLE"]
+    where
+        - HH is hours
+        - MM is minutes
+        - SS is seconds and optional
+        - MS is milliseconds and optional
+        - EXAMPLE is the State String
+
+    All members of tod_group are required to have a tod_state metadata entry.
+    Only static Items require the start_time metadata entry.
+
+    This Rule triggeres at system started, one minute after midnight, and if any
+    member of tod_group_ephem changes. When the Rule triggers, it creates timers
+    to go off at the indicated times.
+
+    For example, they full set of Groups and Items for a weekend, weekday and
+    standard bank holiday system might be something like the following:
+
+        Group Ephem_TimeOfDay_StartTimes
+        Group Ephem_Weekday (Ephem_TimeOfDay_StartTimes) { ToD="day"[type="weekday"] }
+        Group Ephem_Weekend (Ephem_TimeOfDay_StartTimes) { ToD="day"[type="weekend"] }
+        Group Ephem_Holiday (Ephem_TimeOfDay_StartTimes) { ToD="day"[type="holiday] }
```

Comment:
Missing " after holiday.

Reply:
Further development on this PR is blocked until it gets reviewed by a maintainer. In the mean time I've re-implemented this design pattern at https://github.com/rkoshak/openhab-rules-tools. It's significantly different in configuration and use (simpler I hope).